server/eclipse-project/src/main/java/us/freeandfair/corla/model/AuditBoardDashboard.java

```diff
@@ -88,44 +133,57 @@ public void setMembers(final Collection<Elector> the_members) {
   }
   
   /**
-   * Define the CVRs to audit. This clears the set of previously submitted
-   * aCVRs (but they are all retained in persistent storage).
+   * Define the CVRs to audit. This also clears the list of submitted
+   * audit CVRs.
    * 
-   * @param the_cvrs_to_audit The CVRs to audit.
+   * @param the_cvrs_to_audit A list of the IDs of the CVRs to audit,
+   * in the order they should be examined. It must contain no duplicates.
+   * @exception IllegalArgumentException if the list contains duplicates.
    */
-  public void setCVRsToAudit(final Collection<CastVoteRecord> the_cvrs_to_audit) {
-    my_cvrs_to_audit.clear();
-    for (final CastVoteRecord cvr : the_cvrs_to_audit) {
-      my_cvrs_to_audit.put(cvr, null);
-      my_remaining_cvrs.add(cvr);
+  public void setCVRsToAudit(final List<Long> the_cvrs_to_audit) 
+      throws IllegalArgumentException {
+    final Set<Long> duplicate_check = new HashSet<Long>(the_cvrs_to_audit);
+    if (duplicate_check.size() < the_cvrs_to_audit.size()) {
+      throw new IllegalArgumentException("duplicate elements in audit cvr list");
     }
+    my_cvrs_to_audit = new ArrayList<Long>(the_cvrs_to_audit);
+    my_submitted_audit_cvrs.clear();
   }
   
   /**
-   * Submit an aCVR for a CVR under audit.
+   * Submit an audit CVR for a CVR under audit.
    * 
    * @param the_cvr_under_audit The CVR under audit.
-   * @param the_acvr The corresponding aCVR.
-   * @exception IllegalArgumentException if the specified CVR under audit
-   *  is not, in fact, under audit.
+   * @param the_audit_cvr The corresponding audit CVR.
+   * @return true if the audit CVR is submitted successfully, false if it doesn't
+   * correspond to the CVR under audit, or the specified CVR under audit was
+   * not in fact under audit.
    */
-  public void submitAuditCVR(final CastVoteRecord the_cvr_under_audit,
-                             final CastVoteRecord the_acvr) {
-    if (my_cvrs_to_audit.keySet().contains(the_cvr_under_audit)) {
-      my_cvrs_to_audit.put(the_cvr_under_audit, the_acvr);
-    } else {
-      throw new IllegalArgumentException("attempt to set aCVR for unaudited CVR");
+  //@ require the_cvr_under_audit != null;
+  //@ require the_acvr != null;
+  public boolean submitAuditCVR(final CastVoteRecord the_cvr_under_audit, 
+                                final CastVoteRecord the_audit_cvr) {
+    // performs a sanity check to make sure the CVR under audit and the ACVR
+    // are the same card
+    boolean result = true;
+    final int index = my_cvrs_to_audit.indexOf(the_cvr_under_audit.id());
+    if (index > 0) {
```

Comment:
Also, suppose the ID doesn't appear in `my_cvrs_to_audit`. Then `index` is `-1`, so we return the default value of `result`, which is `true`. But we want to return `false` in that case, right?